pkg/cmd/roachtest/tests/tpcc.go

```diff
@@ -410,10 +409,10 @@ func maxSupportedTPCCWarehouses(
 // by the mixed-version framework which chooses a random predecessor version
 // and upgrades until it reaches the current version.
 func runTPCCMixedHeadroom(ctx context.Context, t test.Test, c cluster.Cluster) {
-	maxWarehouses := maxSupportedTPCCWarehouses(*t.BuildVersion(), c.Cloud(), c.Spec())
-	headroomWarehouses := int(float64(maxWarehouses) * 0.7)
-	if c.IsLocal() {
-		headroomWarehouses = 10
+	headroomWarehouses := 10
+	if !c.IsLocal() {
```

Comment:
Nit: the original version is arguably better because it makes the common case "straight line" code. With the change, you now have to branch into `!c.isLocal()` to see how `headroomWarehouses` is derived. If the goal was to get rid of `maxWarehouses` on the main path, we could just replace it by its RHS; don't believe that hurts readability.

Reply:
Oh, is that because `maxSupportedTPCCWarehouses` can't be used when `c.IsLocal()`?

Reply:
>maxSupportedTPCCWarehouses can't be used when c.IsLocal()

Yeah, I was going to make `maxSupportedTPCCWarehouses` support local and just return 15 or something, but there is a test written to make sure it panics when it is local. I'm not opposed to removing that test though, unless I'm missing why we can't support local.

Reply:
> I'm not opposed to removing that test though, unless I'm missing why we can't/don't want to.

Let's remove it then. I'd also update `maxSupportedTPCCWarehouses` to add the explicit case for `c.IsLocal`.

Reply:
Sounds good, done. Seems like the TPCC tests run fine on local, although it's probably only somewhat useful as a dry run.